src/components/leaderboards/Leaderboard/index.js

```diff
@@ -51,95 +51,24 @@ module.exports = React.createClass({
     return {
       isLoading: false,
       boardData: [],
-      currentPage: 1
+      currentPage: 1,
+      childWidth: this.props.childWidth,
     };
   },
 
-  componentWillMount: function() {
-    this.loadLeaderboard();
+  componentDidMount: function() {
+    addEventListener(window, 'resize', this.setChildWidth);
   },
```

Comment:
Fire this.setChildWidth() after you've attached the listener. Otherwise if they don't resize, the childwidth won't be calculated.